tests/main/snap-services/task.yaml

```diff
@@ -17,3 +23,11 @@ execute: |
 
     snap services test-snapd-service > test-snapd-service.out
     MATCH '^test-snapd-service.test-snapd-service\s+ enabled\s+ active\s+ -$' < test-snapd-service.out
+
+    if tests.session has-system-systemd-and-dbus; then
+        # Only run this part of the test on systems supporting systemd
+        # activation n the D-Bus system bus
+        snap install --edge test-snapd-dbus-service
+        snap services test-snapd-dbus-service > dbus-service.out
+        MATCH '^test-snapd-dbus-service.system\s+ disabled\s+ (in)?active\s+ dbus-activated$' < dbus-service.out
```

Comment:
is systemd marking such services as disabled? bit confused

Reply:
confusingly yes I think so:
```
$ systemctl is-enabled snap.test-snapd-dbus-service.system
disabled
```

Reply:
The root cause here is that we still generate an `[Install]` section for some units that we don't enable.  We don't enable socket, timer, or D-Bus activated services:

https://github.com/snapcore/snapd/blob/bf0067c2b25b55e4ca30c778aed68df8410981df/wrappers/services.go#L236-L238

... but we only suppress the `[Install]` section for socket activated services:

https://github.com/snapcore/snapd/blob/bf0067c2b25b55e4ca30c778aed68df8410981df/wrappers/services.go#L730-L734

With the `[Install]` section deleted, systemctl should start reporting the service as `static`, which would cause `snap services` to start reporting it as enabled.  I think this would probably make most sense to address as a separate PR though.

Reply:
We should definitely be consistent with socket actived one here, it seems also less confusing to see actived in our output

Reply:
@jhenstridge so the remedy here is to stop generating `[Install]` sections for timers and D-Bus activated services?

Reply:
@anonymouse64: yes.  I'm happy to fix this up, but suspect it will have knock-on effects on a number of other spread tests.  That's why I suggested doing it as it's own PR.

Reply:
yes that's fine of course to do in a followup PR, thanks for digging into why that happens

Reply:
That followup PR can be found here: https://github.com/snapcore/snapd/pull/9906 -- turned out to be a lot shorter than I thought it might be.